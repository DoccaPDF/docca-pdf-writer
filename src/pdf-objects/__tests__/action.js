import Action from '../action';

const uriAction = Action({
  S: 'URI', URI: 'http://docca.io'
});

const gotoAction = Action({
  S: 'GoTo', D: 'myDest'
});

describe('pdf-objects', () => {
  describe('URI Action', () => {
    it('creates a new object', () => {
      expect(uriAction).toMatchSnapshot();
    });

    it('flattens keys values', () => {
      expect(uriAction.flattenKeys()).toMatchSnapshot();
    });

    it('returns a PDF object', () => {
      expect(uriAction.toPDF()).toMatchSnapshot();
    });
  });

  describe('GoTo Action', () => {
    it('creates a new object', () => {
      expect(gotoAction).toMatchSnapshot();
    });

    it('flattens keys values', () => {
      expect(gotoAction.flattenKeys()).toMatchSnapshot();
    });

    it('returns a PDF object', () => {
      expect(gotoAction.toPDF()).toMatchSnapshot();
    });
  });
});
