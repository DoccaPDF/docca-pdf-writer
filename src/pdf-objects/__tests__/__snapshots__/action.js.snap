// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pdf-objects GoTo Action creates a new object 1`] = `
Object {
  "D": "myDest",
  "S": "GoTo",
}
`;

exports[`pdf-objects GoTo Action flattens keys values 1`] = `
Object {
  "D": "(myDest)",
  "S": "/GoTo",
  "Type": "/Action",
  "URI": undefined,
}
`;

exports[`pdf-objects GoTo Action returns a PDF object 1`] = `
"<<
/S /GoTo
/D (myDest)
>>
"
`;

exports[`pdf-objects URI Action creates a new object 1`] = `
Object {
  "S": "URI",
  "URI": "http://docca.io",
}
`;

exports[`pdf-objects URI Action flattens keys values 1`] = `
Object {
  "D": undefined,
  "S": "/URI",
  "Type": "/Action",
  "URI": "(http://docca.io)",
}
`;

exports[`pdf-objects URI Action returns a PDF object 1`] = `
"<<
/S /URI
/URI (http://docca.io)
>>
"
`;
